src/main/java/io/qameta/allure/bamboo/AllureReportServlet.java

```diff
@@ -66,21 +66,55 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response)
                         }
                     });
                 } else {
-                    final String errorMessage = isEmpty(uploadResult.getFailureDetails()) ?
-                            "Unknown error has occurred during Allure Build. Please refer the server logs for details." :
-                            "Something went wrong with Allure Report generation. Here are some details: \n" + uploadResult.getFailureDetails();
-                    try {
-                        response.setHeader("Content-Type", "text/plain");
-                        response.setHeader("Content-Length", String.valueOf(errorMessage.length()));
-                        response.setHeader("Content-Disposition", "inline");
-                        response.getWriter().write(errorMessage);
-                    } catch (IOException e) {
-                        LOGGER.error("Failed to render error of Allure Report build ", e);
-                    }
+                    uploadResultIsNotSuccess(response, uploadResult);
                 }
             });
         } else {
             LOGGER.info("Path {} does not match pattern", request.getRequestURI());
         }
     }
+
+    @Override
+    protected void doHead(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException {
+        final Matcher matcher = URL_PATTERN.matcher(request.getRequestURI());
+        if (matcher.matches()) {
+            response.setHeader("X-Frame-Options", "ALLOWALL");
+            final String planKeyString = matcher.group(1);
+            final String buildNumber = matcher.group(2);
+            final String filePath = matcher.group(3);
+            final PlanResultKey planResultKey = getPlanResultKey(planKeyString, parseInt(buildNumber));
+            ofNullable(resultsSummaryManager.getResultsSummary(planResultKey)).ifPresent(results -> {
```

Comment:
can we reduce complexity of this method?

Reply:
In later releases.